### ISSUE TITLE
java.lang.NoSuchMethodError: 'javax.crypto.CipherSpi javax.crypto.Cipher.getCurrentSpi()'

### ISSUE THREAD

**Issue by alwa** [robolectric/robolectric#8242]
### Description
When I execute the following source code I get a crash:

```kotlin
        val biometricPrompt = BiometricPrompt(
            activity, executor,
            BiometricsDataRetrievalHandler(encryptedData, callback)
        )
        val promptInfo = PromptInfo.Builder()
            .setTitle(title)
            .setNegativeButtonText(activity.getString(R.string.action_cancel))
            .build()
        biometricPrompt.authenticate(promptInfo, BiometricPrompt.CryptoObject(cipher)) // Here is the crash
``` 

### Steps to Reproduce
@Config(sdk = [30])
@RunWith(AndroidJUnit4::class

### Robolectric & Android Version

- Android 11
- Robolectric 4.9.2

**Comment by utzcoz** [comment]
I can't find this API in OpenJDK: https://github.com/search?q=repo%3Aopenjdk%2Fjdk%20getCurrentSpi&type=code, and it was added by Android JDK team: https://cs.android.com/android/platform/superproject/+/master:libcore/ojluni/src/main/java/javax/crypto/Cipher.java;l=2664-2674?q=Cipher.java&ss=android. @hoisie IIRC, the Robolectric has a method to incept these Android special JDK APIs to make something work?

**Comment by hoisie** [comment]
Yeah I think an entry needs to be added to https://github.com/robolectric/robolectric/blob/master/sandbox/src/main/java/org/robolectric/interceptors/AndroidInterceptors.java. that bytecode will have to be rewritten.

**Comment by utzcoz** [comment]
@alwa Would you like to help us to send a PR to fix this issue. There are samples from https://github.com/robolectric/robolectric/issues/8242#issuecomment-1564519073.

**Comment by alwa** [comment]
@utzcoz Ready for verification. It looks like someone needs to approve the workflows so that they can be run

**Comment by utzcoz** [comment]
Fixed by https://github.com/robolectric/robolectric/pull/8250.

**Comment by alwa** [comment]
Still an issue after my fix. I'll create a new PR

**Comment by alwa** [comment]
> I recall the merged commit at #8262. @alwa you can follow your style to try to fix it with interceptors, and send the new PR with full changes. After some thoughts, instead of intercepting `Cipher`, you can also shadow any methods in the following callstack:
> 
> ```
> java.lang.NoSuchMethodError: 'javax.crypto.CipherSpi javax.crypto.Cipher.getCurrentSpi()'
> 	at android.security.keystore.AndroidKeyStoreProvider.getKeyStoreOperationHandle(AndroidKeyStoreProvider.java:176)
> 	at android.hardware.biometrics.CryptoObject.getOpId(CryptoObject.java:95)
> 	at android.hardware.biometrics.BiometricPrompt.authenticateInternal(BiometricPrompt.java:937)
> 	at android.hardware.biometrics.BiometricPrompt.authenticate(BiometricPrompt.java:865)
> ```
> 
> The above callstack is copied from [#8250 (comment)](https://github.com/robolectric/robolectric/pull/8250#issuecomment-1573298274).

I've started looking into testing this and I now have a failing test... I tried the shadowing approach to fix the problem but then I get this:

![image](https://github.com/robolectric/robolectric/assets/1206247/998a1ecf-68b3-4b79-9f2a-ba1713fbd323)

**Comment by utzcoz** [comment]
Fixed by https://github.com/robolectric/robolectric/pull/8259.
